components/style/properties/shorthand/text.mako.rs

```diff
@@ -5,17 +5,25 @@
 <%namespace name="helpers" file="/helpers.mako.rs" />
 
 <%helpers:shorthand name="text-decoration"
-                    sub_properties="text-decoration-color
-                                    text-decoration-line
-                                    text-decoration-style"
-                    products="gecko"
-                    disable_when_testing="True"
+                    sub_properties="text-decoration-line
+                    ${' text-decoration-style text-decoration-color' if product != 'servo' or data.testing else ''}"
                     spec="https://drafts.csswg.org/css-text-decor/#propdef-text-decoration">
-    use cssparser::Color as CSSParserColor;
-    use properties::longhands::{text_decoration_color, text_decoration_line, text_decoration_style};
+
+    % if product != "servo" or data.testing:
```

Comment:
There are only two products right now. So, let's make this (and all other occurrences) more specific? (like `product == 'gecko'`)